### DISCUSSION TITLE
Avalonia TTS

### DISCUSSION THREAD

**Discussion by YSW33** [AvaloniaUI/Avalonia#15083] [Ideas]
### Is your feature request related to a problem? Please describe.

 Android.Speech.Tts.TextToSpeech

### Describe the solution you'd like

Avalonia have access to Android.Speech.Tts.TextToSpeech  ？

**Comment by thevortexcloud** [comment]
Xamarin/Maui apparently has a Text to Speech API. You should be able to use them in Avalonia app. 


https://learn.microsoft.com/en-us/xamarin/essentials/text-to-speech?tabs=ios

https://learn.microsoft.com/en-us/dotnet/maui/platform-integration/device-media/text-to-speech?view=net-maui-8.0&tabs=android

There is a short guide here on how to use Maui Essentials from within Avalonia:

https://github.com/AvaloniaUI/AvaloniaMauiHybrid

**Reply by YSW33** [reply]
Thank you very much for resolving the issue

https://learn.microsoft.com/en-us/xamarin/essentials/text-to-speech?tabs=android

**Reply by thevortexcloud** [reply]
You are welcome! Make sure you mark the question as answered.

Anyway, to further elaborate on my previous answer for future people: Avalonia uses Xamarin for a lot of it's native integration on mobile. The Xamarin APIs themselves are just wrappers around the Android/iOS APIs (in androids case they map directly to the Java API in most cases).

**Comment by YSW33** [comment]
> Xamarin/Maui apparently has a Text to Speech API. You should be able to use them in Avalonia app.
> 
> https://learn.microsoft.com/en-us/xamarin/essentials/text-to-speech?tabs=ios
> 
> https://learn.microsoft.com/en-us/dotnet/maui/platform-integration/device-media/text-to-speech?view=net-maui-8.0&tabs=android
> 
> There is a short guide here on how to use Maui Essentials from within Avalonia:
> 
> https://github.com/AvaloniaUI/AvaloniaMauiHybrid
